### PR TITLE
Log storage size stats at hash calc

### PR DESCRIPTION
**Pull request by jeffwashington**
#### Problem

new features coming could drastically change the distribution of sizes of storages during runtime and in snapshots.

#### Summary of Changes

Fixes #

### CODE REVIEW

**Comment by jeffwashington** [comment]
devnet stats:
```
storage_size_min=4096i storage_size_quartile_1=40960i storage_size_quartile_2=53248i storage_size_quartile_3=167936i storage_size_max=19038208i storage_size_avg=218593i
```

File: runtime/src/accounts_db.rs [hunk 1]
```diff
@@ -6626,6 +6627,8 @@ impl AccountsDb {
                                             .cloned()
                                             .collect::<Vec<_>>();
                                         if !storages.is_empty() {
+                                            // metrics indicate the collect above over-allocates. This is almost always size 1
+                                            storages.shrink_to_fit();
```
Comment (brooksprumo) [inline runtime/src/accounts_db.rs:6631]: Are you looking at some system memory-related metrics? I went looking for how Vec does its default sizing (and/or any small-vec optimizations). Looks like we'll get [a vec with memory for 4 elements for normal uses of Vec](https://doc.rust-lang.org/src/alloc/raw_vec.rs.html#113-114). I do not know what happens with `.collect()` though. Did your metrics indicate a benefit for this `.shrink_to_fit()`?

**Review by brooksprumo — COMMENTED** [review]
Looks good; had one question

File: runtime/src/accounts_db.rs (hunk 1 above)
Comment (jeffwashington) [reply runtime/src/accounts_db.rs]: I removed it. I don't have time to look at it and it isn't that big of a deal. My metrics were initially 1 level too low and they returned that every slot's vector was size 4 when there was always only 1 item in it. It just offended my sense of right and wrong. But, it is more trouble than it is worth.

**Review by brooksprumo — APPROVED** [review]

**Pull request merged**